eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Genesis.hs

```diff
@@ -219,11 +221,20 @@ validateCostModel (lang, cmps) = case mkCostModel lang cmps of
   Left err -> fail err
   Right cm -> pure (lang, cm)
 
+-- | The keys the Plutus V1 cost models have changed since the Alonzo genesis file was created.
+-- The number of keys, and the ordering of the keys, however, has not changed.
```

Comment:
Why are you so sure the ordering of the keys is the same if "the keys of Plutus V1 cost models have changed"? I'm not saying you are wrong here, but this seems to be DANGEROUS! Would spelling out the mapping (of the differences) be an option, albeit verbose?

Reply:
you are right, there is no guarantee, this is entirely an unenforced and unchecked agreement with the Plutus team. (any change in the ordering would almost certainly cause us to not be able to sync the chain, but this is not very satisfactory at all).  The names themselves are a leaked abstraction that we now regret. The idea with this PR was that the Plutus team could continue to change the names as often as they like, provided they did not change the number or the ordering. If we provided the translation, we would have to maintain the translation, and the abstraction is still leaked.

Reply:
I can, at the very least, add a unit test that checks that the number and the ordering do not change.

Reply:
We're going to replace the textual keys with an opaque enumerable key type with a tightly controlled ordering to make this more explicit.

We didn't expect to have this ordering constraint, hence why it's haphazard currently.